frontend/src/bundles/studio/helpers/get-total-time.ts

```diff
@@ -0,0 +1,12 @@
+type HasDuration = { duration: number };
+
+const calculateTotalMilliseconds = (
+    items: HasDuration[],
+    rangeEnd: number,
+): number => {
+    const totalSeconds = items.reduce((sum, item) => sum + item.duration, 0);
+    const totalMilliseconds = totalSeconds * 1000;
+    return totalMilliseconds > rangeEnd ? totalMilliseconds : rangeEnd;
+};
```

Comment:
```suggestion
const calculateTotalMilliseconds = <T extends { duration: number }>(
    items: T[],
    rangeEnd: number,
): number => {
    const totalSeconds = items.reduce((sum, item) => sum + item.duration, 0);
    const totalMilliseconds = secondsToMilliseconds(totalSeconds);
    return totalMilliseconds > rangeEnd ? totalMilliseconds : rangeEnd;
};
```
I guess it will work

Reply:
Thank you. This is cleaner and I learned about the T type.